src/services/course-service.ts

```diff
@@ -23,6 +25,17 @@ export interface ResourceData {
 export const CourseService = {
   getCourses: async (params?: GetCoursesParams): Promise<AxiosResponse> =>
     await axiosClient.get(URLs.courses.get, { params }),
+
+  getCoursesWithBaseService: (params?: GetCoursesParams) => {
```

Comment:
Params are not optional here
```suggestion
  getCoursesWithBaseService: (params: GetCoursesParams) => {
```

Reply:
Why not optional?

Reply:
Why did you delete params? I meant that params is required for that method, because you should always add some params for searching courses (probably sorting and all that staff). Params were optional because useAxios problem with types.